app/main/controllers/items.py

```diff
@@ -0,0 +1,125 @@
+import json
+
+from flask import request, abort, jsonify
+from sqlalchemy import String, BOOLEAN
+
+from ..models.models import Project, Source, Item
+from ..auth import requires_auth, AuthError
+
+
+def get_authorized_item(user_id, item_id):
+    item = Item.query.get(item_id)
+    if item is None:
+        abort(404)
+
+    if item.project.user_id != user_id:
```

Comment:
I'm not sure if this is the right way to go about this. Not all projects will be directly linked to a project (it might just be the child of a cluster, which itself could be under a project or another cluster (eventually we would reach a project). I'm not sure if Vishnu fixed/will fix this, but we should probably write a function that recursively finds the project of an item or cluster, in case it's not the direct child of a project.

Reply:
Hmm, will definitely get in touch with vishnu and resolve this